backend/src/main/java/de/neuefische/backend/service/EventHubUserDetailService.java

```diff
@@ -36,14 +41,38 @@ public EventHubUserDTO saveUser(UserLoginAndRegistrationDTO user) {
                 .id(generateUUIDService.generateUUID())
                 .username(user.getUsername())
                 .password(passwordEncoder.encode(user.getPassword()))
+                .preferredCategories(new ArrayList<>())
                 .build();
 
-        if(userRepository.findEventHubUserByUsername(newUser.getUsername()).isPresent()) {
+        if (userRepository.findEventHubUserByUsername(newUser.getUsername()).isPresent()) {
             throw new IllegalArgumentException("User with username " + user.getUsername() + " already exists");
         }
 
         EventHubUser temp = userRepository.save(newUser.withRoles(Collections.singletonList(new SimpleGrantedAuthority("user"))).withPassword(passwordEncoder.encode(user.getPassword())));
```

Comment:
temp is not a good name. Name it in precisely the way it is supposed to function.